liota/lib/transports/mqtt.py

```diff
@@ -271,6 +278,7 @@ def connect_soc(self):
         self._paho_client.connect(host=self.url, port=self.port, keepalive=self.keep_alive)
 
         # Start network loop to handle auto-reconnect
+        self._paho_client.reconnect_delay_set(random.randint(1, 15), random.randint(150, 250))
```

Comment:
This is too random for us to understand the behaviour; isn't paho going to slowly move between min and max, anyway, so we should just say 1, 250? Otherwise we need a good comment explaining why we're making min & max random values.

Reply:
When connection is lost, wait initially min_delay seconds and double this time every attempt. The wait is capped at max_delay as per the documentation in Paho MQTT Library.

I have added random delay so that after the max_delay limit is reached every client retries at the different max_delay time.

Reply:
You don't think that the various starting times of trying to make the failed connection by each individual client will be random enough? They're not all operating in-sync...

Reply:
As per QE testing, they figured out that the MQTT broker rejects client connection as all of them try to connect together (due to a large number of DB calls for ACL authorization, it wasn`t occurring before ACL was implemented). This issue should be resolved at database front but I`m trying to add some randomness for every client as they try to re-connect.

Reply:
Again, don't you think there will be enough randomness across clients from when they first try to connect (and fail), and how quickly every subsequent attempt occurs? (I think there is enough randomness there, especially as you back off to almost 5 minutes between retries!) This sounds like you're fixing the QE environment, and not the customer environment. I bet this is a scale test where everything starts very quickly and closely together? I agree we need to support that, but does that apply to a customer environment and should they be penalized with these long back offs?

Reply:
I agree that's the reason I have asked for completing testing report from QE pointing out if this is the main cause. I'll remove the reconnect_delay_set temporarily and just have a random sleep if the disconnection happens unexpectedly.